packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -36,6 +32,19 @@ import { StableRecordIdentifier } from '../../ts-interfaces/identifier';
 import { internalModelFactoryFor, setRecordIdentifier } from '../store/internal-model-factory';
 import CoreStore from '../core-store';
 import coerceId from '../coerce-id';
+import recordDataFor from '../record-data-for';
+
+type DefaultRecordData = import('@ember-data/record-data/-private').RecordData;
+
+function relationshipsFor(instance) {
```

Comment:
I recognize the problem you have down inside `recordDataFor` at the moment (though it *may* be possible for you to work around that with something like what I'm suggesting in the main review comment)—but I wonder if we could constrain the argument and return types *here* not to be implicitly `any`, so that at least callers of *this* will be enabled to do the right thing?